### PR TITLE
chore: remove all remaining legacy analytics native code

### LINKED ISSUE #8795: feat: remove all remaining legacy analytics native code
### What is this about?

Since migration to new MetaMetrics using segment is done, we can remove all Mixpanel analytics native code and dependencies.

### Scenario

_No response_

### Design

_No response_

## Android
- [x] Remove mixpanel native dependencies in android/app/build.gradle
- [x] all native module code in Android related to RCTAnalytics in android/app/src/main/java/io/metamask/nativeModules
- [x] MM_MIXPANEL_TOKEN in .android.env and gradle files

## iOS
- [x] Remove mixpanel native dependencies in ios/mixpanel-iphone and ios/Mixpanel.framework
- [x] all native module code in iOS related to RCTAnalytics in ios/MetaMask/NativeModules/RCTAnalytics
- [x] all mixpanel code in ios
- [x] all framework link in `ios/MetaMask.xcodeproj`
- [x] `ios/mixpanel-iphone` git submodule
- [x] `MM_MIXPANEL_TOKEN` in plist files

## Common
- [x] all mixpanel properties (search for `mixpanel` in all the project)
- [x] remove remapping `MM_MIXPANEL_TOKEN` in scripts/build.sh

## Bitrise
- [ ] #8845

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- no Mixpanel code remaining

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

Remove all Mixpanel analytics native code and dependencies.

### Android
- [x] Remove mixpanel native dependencies in android/app/build.gradle
- [x] all native module code in Android related to RCTAnalytics in android/app/src/main/java/io/metamask/nativeModules
- [x] MM_MIXPANEL_TOKEN in .android.env and gradle files

### iOS
- [x] Remove mixpanel native dependencies in ios/mixpanel-iphone and ios/Mixpanel.framework
- [X] all native module code in iOS related to RCTAnalytics in ios/MetaMask/NativeModules/RCTAnalytics
- [x] all mixpanel code in ios
- [x] all framework link in `ios/MetaMask.xcodeproj`
- [x] `ios/mixpanel-iphone` git submodule
- [x] `MM_MIXPANEL_TOKEN` in plist files

### Common
- [x] all mixpanel properties (search for `mixpanel` in all the project)
- [x] update `remapFlaskEnvVariables` mapping of `MM_MIXPANEL_TOKEN` in `scripts/build.sh`
- [ ] ~update `injectHomePageScripts` in `app/components/Views/BrowserTab/index.js`~

>[!NOTE]
> update `injectHomePageScripts` in `app/components/Views/BrowserTab/index.js` will be done in another PR as it's not mission critical and requires answers form dapp page team.
> see https://github.com/MetaMask/metamask-mobile/issues/8843

>[!NOTE]
> `.github/workflows/security-code-scanner.yml` uses a Mixpanel tocken that is different from the mobile app one. So no impact on it for the moment except that they have to update on their side as we will terminate the Segment subscription. I notified @NicholasEllul.

>[!NOTE]
> TODO added in code related to https://github.com/MetaMask/metamask-mobile/issues/8833
> todo is an accepted issue in Sonar as there's an issue linked to it.

## **Related issues**

Fixes #8795

## **Manual testing steps**

NA

## **Screenshots/Recordings**

NA

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
 /* Begin PBXBuildFile section */
 		07CBADD9D4B441008304F8D3 /* EuclidCircularB-Light.otf in Resources */ = {isa = PBXBuildFile; fileRef = A98029A3662F4C1391489A6B /* EuclidCircularB-Light.otf */; };
 		0C119A06A4353DD11451F541 /* libPods-MetaMask-Flask.a in Frameworks */ = {isa = PBXBuildFile; fileRef = FA078AF57878ABB05D063C8F /* libPods-MetaMask-Flask.a */; };
-		0FD509E0336BF221F6527B24 /* BuildFile in Frameworks */ = {isa = PBXBuildFile; };
+		0FD509E0336BF221F6527B24 /* (null) in Frameworks */ = {isa = PBXBuildFile; };
```
Comment (Cal-L) [inline ios/MetaMask.xcodeproj/project.pbxproj:12]: Can you try cleaning your build folder and rebuilding after removing the Mixpanel files to see if this gets removed entirely?

File: scripts/ios/bundle-js-and-sentry-upload.sh
Comment (Cal-L) [inline scripts/ios/bundle-js-and-sentry-upload.sh:1]: Is it easy to remove this file change? I see this changing in different PRs. I will follow up with addressing this.

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: scripts/ios/bundle-js-and-sentry-upload.sh
Comment (NicolasMassart) [reply scripts/ios/bundle-js-and-sentry-upload.sh:1]: It actually changes only the files access rights. Not sure why.

**Review by Cal-L — DISMISSED** [review]
LGTM

**Review by sethkfman — APPROVED** [review]

**Review by Cal-L — APPROVED** [review]
Lgtm!

**Pull request merged**